cornell-dti/course-plan
# Migrate to Firebase Analytics API

### Summary 
Migrated previous analytics that used the gtag.ts to a Firebase Analytics API. This will hopefully integrate more smoothly with the code base.





This pull request is hopefully going to be a good starting place to begin implementing new analytics tracking in the future

- [x] Changed implementations for GTagEvent and GTagLoginEvent
- [x] Changed GTagEvent and GTagLoginEvent calls to use new logEvent and loginEvent calls

### Test Plan 

- Verify in Firebase backend analytics are appearing.
- Remove a course and test if it is show in backend.

### Notes 

- To implement new analytics tracking, use logEvent as opposed to GTagLogEvent
